util/src/main/java/io/kubernetes/client/informer/cache/Deltas.java

```diff
@@ -0,0 +1,77 @@
+package io.kubernetes.client.informer.cache;
+
+import io.kubernetes.client.util.common.Collections;
+import java.util.List;
+
+/**
+ * Deltas class is the queue'd entry in the DeltaFIFO class. It consists of one or more Delta class
+ * which records the operation type and its state of the object.
+ */
+public class Deltas<ApiType> {
+
+  private List<Delta<ApiType>> deltas;
+
+  public Deltas(List<Delta<ApiType>> deltas) {
+    this.deltas = deltas;
+  }
+
+  public List<Delta<ApiType>> getDeltas() {
+    return deltas;
+  }
+
+  void appendDelta(Delta<ApiType> delta) {
+    this.deltas.add(delta);
+  }
+
+  Delta<ApiType> newest() {
+    if (Collections.isEmptyCollection(deltas)) {
+      return null;
+    }
+    return deltas.get(deltas.size() - 1);
+  }
+
+  Delta<ApiType> oldest() {
+    if (Collections.isEmptyCollection(deltas)) {
+      return null;
+    }
+    return deltas.get(0);
+  }
+
+  public static class Delta<ApiType> {
+
+    private DeltaType type;
+
+    private ApiType obj;
+
+    public Delta(DeltaType type, ApiType obj) {
```

Comment:
Perhaps use [MutablePair<DeltaType, ApiType>](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/tuple/MutablePair.html) instead of a custom class? that way toString, hashCode, etc will all work correctly.

Reply:
done